src/components/_classes/nested/NestedComponent.js

```diff
@@ -643,6 +643,7 @@ export default class NestedComponent extends Field {
   }
 
   checkValidity(data, dirty, row, silentCheck) {
+    row = this.dataValue || row;
```

Comment:
I am afraid, this change can create some regressions in validations of dataGrid/editGrid/containers/custom components with rows where row is not equal to the component dataValue.
Addionally:
1. Row, passed as a parameter, is expected to have first priority. In your case component dataValue is prioritized.
2. logically, row means a context (like environment) where the component, that is being checked, exists. The dataValue is the value of the component itself, not the context.

Checking the 5511 ticket, I see that the issue is in the Container Component. When we are in checkValidity, row value is correct for the component itself but not correct for its children as row for children is expected to be equal to the parent container dataValue. This is very specific to Container and is not related to some other nested components like fieldSet or panel. So, I would prefer to move the nested components validity check to separate method in NestedComponent and then to redefine this method for ContainerComponent and pass container dataValue as a row to it.
Something like this:

**Nested Component**

```
  checkChildComponentsValidity(data, dirty, row, silentCheck, isParentValid) {
    return this.getComponents().reduce(
      (check, comp) => comp.checkValidity(data, dirty, row, silentCheck) && check,
      isParentValid
    );
  }
```

```
  checkValidity(data, dirty, row, silentCheck) {
    if (!this.checkCondition(row, data)) {
      this.setCustomValidity('');
      return true;
    }
    const isValid = this.checkChildComponentsValidity(data, dirty, row, silentCheck, super.checkValidity(data, dirty, row, silentCheck));
    this.checkModal(isValid, dirty);
    return isValid;
  }
```

**ContainerComponent**

```
  checkChildComponentsValidity(data, dirty, row, silentCheck, isParentValid) {
    return super.checkChildComponentsValidity(data, dirty, this.dataValue, silentCheck, isParentValid);
  }
```